scripts/convert.py

```diff
@@ -171,22 +212,35 @@ def check_alignments(self, frame):
                        "skipping".format(frame))
         return have_alignments
 
-    def convert(self, converter, item):
+    ##removed converter, using self.converters
+    def convert(self, item):
         """ Apply the conversion transferring faces onto frames """
         try:
             filename, image, faces = item
-            skip = self.opts.check_skipframe(filename)
-
-            if not skip:
-                for face in faces:
-                    image = converter.patch_image(image, face)
-                filename = str(self.output_dir / Path(filename).name)
-
-                if self.args.draw_transparent:
-                    filename = "{}.png".format(os.path.splitext(filename)[0])
-                    logger.trace("Set extension to png: `%s`", filename)
-
-                cv2.imwrite(filename, image)  # pylint: disable=no-member
+            ##skip = self.opts.check_skipframe(filename)
+
+            ##if not skip:
+            for face in faces:
+                ## pick best model and swap
+                im = Image.fromarray(face.image)
+                try:
+                    unknown_face_encoding = face_recognition.face_encodings(np.array(im))[0]
+                    results = face_recognition.face_distance(self.target_encodings, unknown_face_encoding)
+                    self.matched_face = np.argmin(results)
+                    print(self.matched_face)     
+                except:
```

Comment:
a face encoding for every image and a model for every image seems like overkill. Face_recognition has a k_means clustering technique to consolidate multiple encodings of the same person / identity into a master encoding.  You could argmin against this master and then would only need one model per identity then?

Reply:
could potentially cluster all of the images in the train set to create an encoding, should be an insignificant amount of time compared to training the nn - although as long as faces are different enough selecting which face you've detected seems to pick the right one pretty accurately.